relay-general/src/processor/selector.rs

```diff
@@ -221,15 +225,12 @@ impl FromStr for SelectorSpec {
             _ => {}
         }
 
-        handle_selector(
-            SelectorParser::parse(Rule::RootSelector, s)
-                .map_err(|e| InvalidSelectorError::ParseError(Box::new(e)))?
-                .next()
-                .unwrap()
-                .into_inner()
-                .next()
-                .unwrap(),
-        )
+        let rule_pair = match SelectorParser::parse(Rule::RootSelector, s) {
+            Ok(mut rule_pairs) => rule_pairs.next().unwrap().into_inner().next().unwrap(),
+            Err(e) => return Ok(SelectorSpec::Other(e.to_string())),
```

Comment:
I need some input on whether this logic makes sense at all

Reply:
This logic seems different to me. Previously, errors were mapped to `InvalidSelectorError`, and in the new diff we are parsing the errors into an `Ok`.